app/scripts/views/settings/gravatar_permissions.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'cocktail',
+  'views/form',
+  'stache!templates/settings/gravatar_permissions',
+  'lib/promise',
+  'views/mixins/back-mixin',
+  'views/mixins/service-mixin'
+],
+function (Cocktail, FormView, Template, p, BackMixin, ServiceMixin) {
+  'use strict';
+
+  var PERMISSIONS = ['profile:email'];
+  var GRAVATAR_MOCK_CLIENT_ID = 'gravatar';
+
+  var View = FormView.extend({
+    template: Template,
+    className: 'gravatar_permissions',
+
+    context: function () {
+      var account = this.getSignedInAccount();
+      var serviceName = this.translator.get('Gravatar');
+      return {
+        email: account.get('email'),
+        serviceName: serviceName
+      };
+    },
+
+    beforeRender: function () {
+      var account = this.getSignedInAccount();
+      if (account.hasGrantedPermissions(GRAVATAR_MOCK_CLIENT_ID, PERMISSIONS)) {
+        this.logScreenEvent('already_accepted');
```

Comment:
This reminds me that we do not log similar events in the [sign in](https://github.com/mozilla/fxa-content-server/blob/26ef3ae90fdd3ac5073a1b1dfb3c84a602ccd19e/app/scripts/views/sign_in.js#L123) and [sign up](https://github.com/mozilla/fxa-content-server/blob/26ef3ae90fdd3ac5073a1b1dfb3c84a602ccd19e/app/scripts/views/sign_up.js#L279) views. The logic is slightly different in those views to decide whether to display the gravatar permissions, the choice is made before the screen is ever displayed.